test/ngResource/resourceSpec.js

```diff
@@ -2026,6 +2143,41 @@ describe('handling rejections', function() {
     }
   );
 
+
+  it('should not propagate exceptions in success callback to the returned promise', function() {
```

Comment:
why is it a bad thing for this to happen?

Reply:
Imo, the success callback should not interfere with the promise chain, since it's strictly just a callback. Error handling should only happen in the interceptors. Do you think this will affect many apps?

Reply:
I agree that in an ideal world this should be the behaviour - I am just wondering if it is so important that it should cause a BC.

Reply:
Let's discuss in https://github.com/angular/angular.js/pull/16446/files#r169286982.